spec/expression.dd

```diff
@@ -21,13 +21,44 @@ full expression of $(I expr$(SUBSCRIPT 1)). Otherwise, $(I expr) is its own full
 
 $(P Each expression has a unique full expression.)
 
-Example: in the statement `return f() + g() * 2;`, the full expression of `g() * 2` is `f() + g() * 2`, but not the
-full expression of `f() + g()` because the latter is not parsed as a subexpression.
+$(P Example: in the statement `return f() + g() * 2;`, the full expression of `g() * 2` is `f() + g() * 2`, but not the
+full expression of `f() + g()` because the latter is not parsed as a subexpression.)
 
-Note: Although the definition is straightforward, a few subtleties exist related to function literals. In the
+$(P Note: Although the definition is straightforward, a few subtleties exist related to function literals. In the
 statement `return (() => x + f())() * g();`, the full expression of `f()` is `x + f()`, not the expression passed
-to `return`. This is because the parent of `x + f()` has function literal type, not expression type.
-
+to `return`. This is because the parent of `x + f()` has function literal type, not expression type.)
+
+$(DDOC_ANCHOR define-lvalue)$(P $(B Definition) ($(DOUBLEQUOTE Lvalue)): The following expressions, and no
+others, are called lvalue expressions or lvalues:)
+$(OL
+$(LI `this` inside `struct` and `union` member functions;)
+$(LI the result of the $(I DotIdentifier) grammatical construct `.` (left side may be missing) when the
+rightmost side of the dot is the name of a field (whether direct or `static`), variable, or function;)
+$(LI the result of the following expressions:
+$(UL
+$(LI built-in unary operators `+` (when applied to an lvalue), `*`, `++` (prefix only), `--` (prefix only);)
```

Comment:
Unary `+` indeed preserves lvalues, but I'm not convinced that it should.

Reply:
@tgehr yah, documenting existing stuff instead of aspiring to perfection

Reply:
On second thought, "+ is a no-op" is easier to explain and understand than "+ is almost a no-op, you see, it copies the thing into an rvalue".

Reply:
No, this makes no sense. Note that soon, + will perform integer promotion, and then promoted lvalues will most likely be rvalues. Much easier to just deprecate using +x as an lvalue completely, as otherwise either your explanation or your justification of the behavior is bound to become a lot more complicated. There is no need to innovate with unary +.